spec/model/compatibility-check/check-model.spec.ts

```diff
@@ -4,158 +4,915 @@ import { ValidationResult } from '../../../src/model/validation/result';
+                        field: String @modules(all: true)
+                    }
+                `,
+                gql`
+                    type Test @rootEntity {
+                        field: [String]
+                    }
+                `,
+            );
+            expectSingleCompatibilityIssue(
+                result,
+                'Field "Test.field" should not be a list (required by module "module1").',
+            );
+        });
     });
 
-    it('rejects if a field has the wrong type', () => {
-        const result = run(
-            gql`
-                type Test @rootEntity @modules(in: "module1") {
-                    field: String @modules(all: true)
-                }
-            `,
-            gql`
-                type Test @rootEntity {
-                    field: Int
-                }
-            `,
-            ['module1'],
-        );
-        expectSingleCompatibilityIssue(
-            result,
-            'Field "Test.field" needs to be of type "String" (required by module "module1").',
-        );
+    describe('@key', () => {
+        it('rejects if a field needs to be a key field', () => {
+            const result = run(
+                gql`
+                    type Test @rootEntity @modules(in: "module1") {
+                        field: String @key @modules(all: true)
+                    }
+                `,
+                gql`
+                    type Test @rootEntity {
+                        field: String
+                    }
+                `,
+            );
+            expectSingleCompatibilityIssue(
+                result,
+                'Field "Test.field" needs to be decorated with @key (required by module "module1").',
+            );
+        });
+        it('accepts if the field is properly decorated with @key', () => {
+            const result = run(
+                gql`
+                    type Test @rootEntity @modules(in: "module1") {
+                        field: String @key @modules(all: true)
+                    }
+                `,
+                gql`
+                    type Test @rootEntity {
+                        field: String @key
+                    }
+                `,
+            );
+            expectToBeValid(result);
+        });
+
+        it('rejects if the id field needs to be decorated with @key but is missing', () => {
+            const result = run(
+                gql`
+                    type Test @rootEntity @modules(in: "module1") {
+                        id: ID @key
+                        field: String @modules(all: true)
+                    }
+                `,
+                gql`
+                    type Test @rootEntity {
+                        field: String
+                    }
+                `,
+            );
+            expectSingleCompatibilityIssue(
+                result,
+                'Field "id: ID @key" needs to be specified (required by module "module1").',
```

Comment:
I was a bit reluctant with using the type + field name everywhere because it makes the messages longer and hides the interesting part. If you look at the message in the editor and hover over the change, you don't need to be told what the field is called because you're looking right at it.

I wonder if we should add an additional property to the message (that could even be automatically generated from the location if it's present) for the context of the message. We could then
- prepend it in textual representations of the message
- format it in a special way in a UI that lists all messages (similar to the source name and line number)
- omit it in the code editor hints
Do you think that would be a good idea? If yes, I would keep the messages as-is and change them into this new format in a later commit.

Reply:
Yes 👍 , I think that I came from the perspective of looking at the validation panel list where more context would definitely be beneficial. So kind of a short/concise message when looking at the error in the editor, and a longer message when looking at the message in a list would be perfect.

> prepend it in textual representations of the message

I do not know exactly what you mean by that

> automatically generated from the location if it's present) for the context of the message

Not sure about the automatic generation, I thought we can simply add another field to the validation message with a longer description